### PR TITLE
Inve 19535 fix nested combo expansion

### PR DESCRIPTION
**Pull request by Verdagio**
##### Checklist

- [x] `npm test` passes

##### Description of change

Expands parent combos when child is expanded

see comments for screencap

### CODE REVIEW

**Comment by Verdagio** [comment]
Some minor changes required to get stuff to sit correctly

File: packages/core/src/global.ts [hunk 1]
```diff
@@ -64,7 +64,7 @@ const colorSet = {
 };
 
 export default {
-  version: '0.7.4-siren.0',
+  version: '0.7.4-siren.1',
```
Comment (Verdagio) [inline packages/core/src/global.ts:67]: fixes a broken test

**Comment by Verdagio** [comment]
Note, repositioning nodes on the graph on combo expansion is pushed out to the hard cut tasks https://sirensolutions.atlassian.net/browse/INVE-19841

File: packages/core/src/graph/controller/item.ts [hunk 2]
```diff
@@ -478,6 +498,12 @@ export default class ItemController {
         graph.showItem(c, stack);
       }
     });
+    
+    if (typeof combo !== 'string') {
+      setTimeout(() => {
+        this.updateParentCombo(combo as ICombo);
+      }, 200);
```
Comment (Verdagio) [inline packages/core/src/graph/controller/item.ts:505]: There's some race condition happening here requiring this action to be delayed

https://user-images.githubusercontent.com/15673201/194109893-7b981b0e-0271-4246-9f4d-3a776f2e726a.mov

File: packages/core/src/graph/controller/item.ts [hunk 3]
```diff
@@ -348,6 +348,28 @@ export default class ItemController {
...
+   */
+  public updateParentCombo(combo: ICombo) {
+    const model = combo.getModel() as ComboConfig;
+    if (model.parentId) {
+      const graph = this.graph;
+      const parent = graph.findById(model.parentId) as ICombo;
+      const parentModel = parent.getModel();
+      const parentBbox = getComboBBox(parentModel.children as ComboTree[], graph, parent);
+
+      parentBbox.width += 10;
```
Comment (Verdagio) [inline packages/core/src/graph/controller/item.ts:364]: adds buffer to prevent child circles running out of bounds

**Comment by Verdagio** [comment]
@daniele-pini @TonyN96 this one is ready now

File: packages/core/src/graph/controller/item.ts [hunk 4]
```diff
@@ -447,6 +442,9 @@ export default class ItemController {
     children.combos.forEach((c) => {
       graph.hideItem(c, stack);
     });
+    setTimeout(() => {
+      this.updateAncestorCombos(combo as ICombo);
+    }, 0);
```
Comment (daniele-pini) [inline packages/core/src/graph/controller/item.ts:447]: You should use this one instead: https://github.com/sirensolutions/G6/blob/2c326f07eb9300c5fd4675c9034270100ccc9cbf/packages/core/src/graph/graph.ts#L1964-L2018

Note how that one also travels up the combo tree hierarchy. And check if that works without the `setTimeout`, too.

**Review by daniele-pini — CHANGES_REQUESTED** [review]

**Review by Blakko — COMMENTED** [review]
Would it be possible to have the child combo to not go outside of the parent?

**Comment by daniele-pini** [comment]
@Blakko, we're revising the approach - turns out there's actually a bug when _making_ groups, which prevents a correct update. When we fix that, children should placed correctly inside their parents.

**Comment by Verdagio** [comment]
> Would it be possible to have the child combo to not go outside of the parent?

@Blakko The recording is out of date, this is current behaviour (as per this pr) note looks like master has a bug requiring an additional click away in order for the last action to happen?

https://user-images.githubusercontent.com/15673201/195130663-daf4a6f6-9815-4e28-8637-c9ac516829aa.mov

**Comment by Blakko** [comment]
> note looks like master has a bug requiring an additional click away in order for the last action to happen?

@Verdagio yes we have a PR in review that will fix that issue

**Comment by daniele-pini** [comment]
Superseded by #50.

**Pull request closed without merging**